apitools/org.eclipse.pde.api.tools/src/org/eclipse/pde/api/tools/internal/model/BundleComponent.java

```diff
@@ -770,41 +763,21 @@ static void extractDirectory(ZipFile zip, String pathprefix, File parent) throws
 	 * @throws IOException
 	 */
 	static File extractEntry(ZipFile zip, ZipEntry entry, File parent) throws IOException {
-		InputStream inputStream = null;
-		File file;
-		FileOutputStream outputStream = null;
-		try {
-			inputStream = zip.getInputStream(entry);
-			file = new File(parent, entry.getName());
+		try (InputStream inputStream = zip.getInputStream(entry)) {
```

Comment:
If you can contribute the effort, please.

Reply:
You have rewritten the whole method already so it would be too hard to use Path API instead...

Reply:
Additionally the check with `if (!file.toPath().normalize().startsWith(parent.toPath().normalize())) {` also looks pretty much the same as in the `extractDirectory()` method above, but since that does not seem to be trivial it could be refactored into a method like 
```
private static getEntryFile(Path parent, String name)
	Path file = parent.resolve(name);
	if (!file.normalize().startsWith(parent.normalize())) {
		throw new ZipException("Bad zip entry: " + name); //$NON-NLS-1$
	}
	return file;
```